packages/exo/test/test-legacy-guard-tolerance.js

```diff
@@ -0,0 +1,140 @@
+// @ts-check
+// eslint-disable-next-line import/order
+import { test } from './prepare-test-env-ava.js';
+
+import { passStyleOf } from '@endo/pass-style';
+import {
+  M,
+  getAwaitArgGuardPayload,
+  getInterfaceGuardPayload,
+  getMethodGuardPayload,
+} from '@endo/patterns';
+import {
+  makeLegacyAwaitArgGuard,
+  makeLegacyMethodGuard,
+  makeLegacyInterfaceGuard,
+} from './make-legacy-guards.js';
+import { makeExo } from '../src/exo-makers.js';
+
+test('legacy guard tolerance', async t => {
```

Comment:
Why not create a `test.failing` case, instead of testing for an expected error?

Reply:
If #2038 was going to take awhile, I would. But since it is about to overwrite the test output, I wanted to leave behind an accurate record of what the state before #2038 was.

Reply:
`test.failing` can also hide failure modes other than the one you want to test for.

Reply:
Stronger reason (that I think was at the back of my mind somewhere):

The test output in this PR is the correct one, i.e., without the kludge adaptors of #2038 that will become irrelevant once we no longer need to support code prior to #1712 . At that point, we should not exactly revert #2038 , but we should revert the effect of #2038 by simplifying getGuardPayloads.js, and we should exactly revert #2038's changes to this PR's test-legacy-guard-tolerance.js, i.e., restore the version in this PR.

Reply:
In any case, seeing the thumbs up, resolving.